pdns/lua-base4.cc

```diff
@@ -42,6 +42,15 @@ BaseLua4::~BaseLua4() { }
 void BaseLua4::prepareContext() {
   d_lw = std::unique_ptr<LuaContext>(new LuaContext);
 
+  // lua features available
+  d_lw->writeVariable("pdns_features",
+                      vector<pair<string, boost::variant<string,bool,int,double> > > {
+  // Add key-values pairs below. Make sure you add string values explicity converted to string.
+  // e.g. { "somekey", string("stringvalue") }
+  // Both int and double end up as a lua number type.
+      { "PR8001_devicename", true },
+    });
+  
```

Comment:
This file is shared with the Auth, so that might not be the right place to define this variable?

Reply:
Or at least, not its contents!

Reply:
From IRC conversation: if we want to share this feature between the different products, maybe the features should be reported into this table at runtime. For example, the one auth 'feature' I can think of right now is 'geoip is available and configured', which we don't know at compile time. Even if the recursor does know everything at compile time, filling the table at startup would make `pdns_features` "portable" between the products.

Alternatively, go with with @rgacogne said and move it into a Recursor-specific file, we can always think about features reporting for Auth (and dnsdist?) later.

Reply:
I adapted the setup. Base class now has the common stuff. It calls a method that can be overriden to define the product specifics values.